README.md

```diff
@@ -1,16 +1,39 @@
 
-# Overview
+* [Overview](#overview)
+    * [`dotnet new`](#dotnet-new)
+    * [Template Content Repositories](#template-content-repositories)
+* [How to author the templates](#how-to-author-the-templates)
+    * [Template Samples](#template-samples)
+* [Contributing](#contributing)
+* [How to build, run & debug](#how-to-build-run--debug)
+* [Trademarks](#trademarks)
 
-This repository is the home for the .NET Core Template Engine. It contains the brains for `dotnet new` and template instantiation in Visual Studio.
-When `dotnet new` is invoked, it will call the Template Engine to create the artifacts on disk.
-Template Engine is a library for manipulating streams, including operations to replace values, include/exclude 
-regions and process `if`, `else if`, `else` and `end if` style statements.
+## Overview
 
-# Template Content Repositories
+This repository is the home for the .NET Template Engine. It contains the libraries for template instantiation  and template package management used in [`dotnet new`](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-new) and [New Project Dialog](https://learn.microsoft.com/en-us/visualstudio/ide/create-new-project?view=vs-2022) and New Item Dialog in Visual Studio and Visual Studio for Mac. The libraries are also distributed as the NuGet packages on nuget.org.
 
-[.NET default templates](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new-sdk-templates) are located in several repositories.
+The key packages are:
+|Package name| Description|
+|---|---|
+| `Microsoft.TemplateEngine.Edge` | The template engine infrastructure:  managing template packages, templates, components, executing template. Main API surface for the products aiming to use template engine. See ['Inside Template Engine'](docs/api/Inside-the-Template-Engine.md) article for more information. |
+| `Microsoft.TemplateEngine.Abstractions` | Contains the main contracts between `Edge` and components |
+| `Microsoft.TemplateEngine.Orchestrator.RunnableProjects` | The template generator based on `template.json` configuration |
+| `Microsoft.TemplateSearch.Common` | Facilitates template packages search on nuget.org |
+| `Microsoft.TemplateEngine.IDE` | Lightweight API overlay over `Microsoft.TemplateEngine.Edge`. |
+| `Microsoft.TemplateEngine.Authoring.Tasks` | Authoring tools: MSBuild tasks for template authoring |
+| `Microsoft.TemplateEngine.Authoring.CLI` | Authoring tools: dotnet CLI tool with utilities for template authoring |
+| `Microsoft.TemplateEngine.Authoring.TemplateVerifier` | Authoring tools: [snapshot testing framework](docs/authoring-tools/Templates-Testing-Tooling.md) for the templates |
+
+### `dotnet new`
+
+`dotnet new` CLI is now located in [dotnet/sdk](https://github.com/dotnet/sdk/tree/main/src/Cli/Microsoft.TemplateEngine.Cli) repo.
+
+The issues for `dotnet new` CLI UX should be opened in the that repository.
 
+### Template Content Repositories
```

Comment:
I wish we could extract this out and reuse it in both documents

Reply:
https://github.com/github/markup/issues/346 +100500 + 1 user would agree with you :)